08-generic_cache/01_naive/cache.go

```diff
@@ -0,0 +1,37 @@
+package cache
+
+// Cache is key, value storage.
+type Cache[K comparable, V any] struct {
+	data map[K]V
+}
+
+// New create a new Cache with an initialised data.
+func New[K comparable, V any]() Cache[K, V] {
+	return Cache[K, V]{
+		data: make(map[K]V),
+	}
+}
+
+// Read returns the associated value for a key,
+// or ErrNotFound if the key is absent.
+func (c *Cache[K, V]) Read(key K) (V, error) {
```

Comment:
I would recommend returning `(V, bool)` which is easier to handle in code rather than with error.

Reply:
Done.

Reply:
Any reason this change was reverted?